src/couch_views/src/couch_views_jobs.erl

```diff
@@ -0,0 +1,122 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_views_jobs).
+
+-export([
+    status/2,
+    add/2,
+
+    accept/0,
+    get_job_data/1,
+    update/5,
+    finish/5,
+    set_timeout/0,
+
+    subscribe/2,
+    wait/1,
+    unsubscribe/1,
+
+    create_job_id/2
+]).
+
+
+-include("couch_views.hrl").
+
+
+% Query request usage of jobs
+
+
+status(TxDb, Mrst) ->
+    JobId = create_job_id(TxDb, Mrst),
+
+    case couch_jobs:get_job_state(TxDb, ?INDEX_JOB_TYPE, JobId) of
+        {ok, State} -> State;
+        {error, not_found} -> not_found;
+        Error -> Error
+    end.
+
+
+add(TxDb, Mrst) ->
+    JobData = create_job_data(TxDb, Mrst, 0),
+
+    JobId = create_job_id(TxDb, Mrst),
+    JTx = couch_jobs_fdb:get_jtx(),
```

Comment:
This will create a new transaction object. We can pass `TxDb` to `get_jtx(TxDb)` and it should use the existing TxDb transaction.